KarhooSDK/Service/Auth/LoginInteractor/KarhooAuthLoginWithCredentialsInteractor.swift

```diff
@@ -13,22 +13,24 @@ final class KarhooAuthLoginWithCredentialsInteractor: AuthLoginWithCredentialsIn
     private let userInfoSender: RequestSender
     private let userDataStore: UserDataStore
     private let analytics: AnalyticsService
-    private let paymentProviderRequest: RequestSender
-    private let loyaltyProviderRequest: RequestSender
-    private let nonceRequestSender: RequestSender
-    
+    private let paymentProviderUpdateHandler: PaymentProviderUpdateHandler
+
     init(userInfoSender: RequestSender = KarhooRequestSender(httpClient: TokenRefreshingHttpClient.shared),
          userDataStore: UserDataStore = DefaultUserDataStore(),
          analytics: AnalyticsService = KarhooAnalyticsService(),
          paymentProviderRequest: RequestSender = KarhooRequestSender(httpClient: JsonHttpClient.shared),
          loyaltyProviderRequest: RequestSender = KarhooRequestSender(httpClient: JsonHttpClient.shared),
-         nonceRequestSender: RequestSender = KarhooRequestSender(httpClient: TokenRefreshingHttpClient.shared)) {
+         nonceRequestSender: RequestSender = KarhooRequestSender(httpClient: TokenRefreshingHttpClient.shared),
+         paymentProviderUpdateHandler: PaymentProviderUpdateHandler? = nil
+    ) {
         self.userInfoSender = userInfoSender
         self.userDataStore = userDataStore
         self.analytics = analytics
-        self.paymentProviderRequest = paymentProviderRequest
-        self.nonceRequestSender = nonceRequestSender
-        self.loyaltyProviderRequest = loyaltyProviderRequest
+        if paymentProviderUpdateHandler != nil {
```

Comment:
This could be written shorter and without force-unwrapping

```
let handler =  KarhooPaymentProviderUpdateHandler(nonceRequestSender: nonceRequestSender, paymentProviderRequest: paymentProviderRequest)
self.paymentProviderUpdateHandler = paymentProviderUpdateHandler ??  handler
```

The same goes for next examples.

Reply:
In your approach we will create `let handler` and in many cases we will not use it. So why waste time and memory? If force unwrapping after if != nil is not ok for you I can change it to `if let`